### PR TITLE
[ Rel-5_0 Bug 10691 ] - No CustomerID shown after TicketCreate and modified Unit test

### PR DESCRIPTION
**Pull request by s7design**
https://bugs.otrs.org/show_bug.cgi?id=10691

Hello @mgruner 

This is proposal for no auto-assigning CustomerID and CustomerUser for unknown incoming emails in accordance with your suggestion. Because of that, if there are no CustomerID and CustomerUser found, parameters with their values which are used later for ticket creating stay undefined.

IF condition is modified because there is the same condition in parent IF and '$GetParam{'X-OTRS-CustomerUser'}'  can not change 
(https://github.com/s7design/otrs/blob/5241e87234e22ba647dac196f57253926e7d18e3/Kernel/System/PostMaster/NewTicket.pm#L187).

Please let me know if you have any suggestion for this PR.

Regards,
Milan

### CODE REVIEW

File: scripts/test/Console/Command/Maint/PostMaster/Read.t [hunk 1]
```diff
@@ -12,15 +12,85 @@ use utf8;
...
+$Self->True(
+    $CustomerUserID,
+    "CustomerUserID $CustomerUserID is created",
+);
+
+my @Tests = (
+    {
+        EmailAddress => 'Unknown' . $CustomerUserEmailAddress,
+        Result       => {
+            CustomerUserID => undef,
```
Comment (mgruner) [inline scripts/test/Console/Command/Maint/PostMaster/Read.t]: Actually this is not the requested behaviour. CustomerUserID should still be filled with the email address to have at least something to identify in the ticket. But CustomerID should stay empty, just like it is if you assign an unknown CustomerUser in AgentTicketCustomer.

**Comment by s7design** [comment]
Hi @mgruner,

Thanks for your follow-up. No problem it will be updated as you suggested.

Regards
Milan

**Comment by s7design** [comment]
Hello @mgruner 

PR is updated in accordance with your suggestions. Please inform me if you have any comment for these changes.

Regards,
Milan

File: Kernel/System/PostMaster/NewTicket.pm [hunk 2]
```diff
@@ -184,7 +184,7 @@ sub Run {
         }
 
         # take CustomerID from customer backend lookup or from from field
-        if ( $CustomerData{UserLogin} && !$GetParam{'X-OTRS-CustomerUser'} ) {
+        if ( $CustomerData{UserLogin} ) {
```
Comment (mgruner) [inline Kernel/System/PostMaster/NewTicket.pm]: I omitted this change, I believe it is still valid.

**Comment by mgruner** [comment]
Merged, thanks!

**Pull request closed without merging**